newsroom/topics_folders/folders.py

```diff
@@ -0,0 +1,89 @@
+from enum import Enum, unique
+from typing import Optional, Annotated
+
+from newsroom import MONGO_PREFIX
+from newsroom.core.resources.model import NewshubResourceModel
+from newsroom.core.resources.service import NewshubAsyncResourceService
+
+from newsroom.topics.topics_async import TopicService
+
+from superdesk.core.web import EndpointGroup
+from superdesk.core.resources.fields import ObjectId as ObjectIdField
+from superdesk.core.resources import (
+    ResourceConfig,
+    MongoIndexOptions,
+    MongoResourceConfig,
+    RestEndpointConfig,
+    RestParentLink,
+)
+from superdesk.core.resources.validators import validate_data_relation_async
+
+
+@unique
+class SectionType(str, Enum):
+    WIRE = "wire"
+    AGENDA = "agenda"
+    MONITORING = "monitoring"
+
+
+class FolderResourceModel(NewshubResourceModel):
+    name: str
+    parent: Annotated[Optional[ObjectIdField], validate_data_relation_async("topic_folders")] = None
+    section: SectionType
+
+
+class FolderResourceService(NewshubAsyncResourceService[FolderResourceModel]):
+    resource_name = "topic_folders"
```

Comment:
No need to define `resource_name` on async services, this will be populated on registration